### PR TITLE
Dev: Add "framework" option

### PR DESCRIPTION
**Pull request by RaeesBhatti**
**- Summary**
Add `framework` option for `netlify.toml` `dev` section. This allows users to specify `#auto` to test all detectors, `#static` to run static file server or one of [project types which Netlify Dev can detect](https://github.com/netlify/cli/tree/master/src/detectors) to run that detector only.
This will help with situations where users have multiple detectors testing positive and they always have to choose.

Fixes: https://github.com/netlify/cli/issues/469
Fixes: https://github.com/netlify/cli/issues/706
Fixes: https://github.com/netlify/cli/issues/841

**- Test plan**
* Set `framework = "create-react-app"` or a [project types which Netlify Dev can detect](https://github.com/netlify/cli/tree/master/src/detectors)
* Run `netlify dev`

**- Description for the changelog**
* Add `framework` option for `netlify.toml`
* Add docs for `framework` option
* Change `type` to `framework` in detectors
* Rename  `startProxy` param `serverType` to `framework`
* Create `loadDetector` function
* Replace `chooseDefaultSettings` function with `chooseDefaultArgs`
* Add default `env: { ...process.env }` to `settings`
* Rename detector file `cra` to `create-react-app`
* Dont slice `run` from command arguments if command is `npm`

**- A picture of a cute animal (not mandatory but encouraged)**
🦄

### CODE REVIEW

**Comment by RaeesBhatti** [comment]
Can we get a review on this soon please.
https://community.netlify.com/t/env-node-no-such-file-or-directory-when-using-netlify-cli/12391

File: src/commands/dev/index.js [hunk 1]
```diff
@@ -577,7 +528,7 @@ DevCommand.flags = {
   live: flags.boolean({
     char: 'l',
     description: 'Start a public live session'
-  })
+  }),
```
Comment (ehmicky) [inline src/commands/dev/index.js:531]: Shouldn't Prettier pick up those?

File: src/utils/detect-server.js [hunk 2]
```diff
@@ -1,99 +1,161 @@
...
         return null
       }
     })
-    .filter(Boolean)
-  let settingsArr = []
-  let settings = null
-  for (const i in detectors) {
-    const detectorResult = detectors[i]()
-    if (detectorResult) settingsArr.push(detectorResult)
+    for (const i in detectors) {
```
Comment (ehmicky) [inline src/utils/detect-server.js:27]: Can also do:

```js
for (const detector of detectors) {
  const detectorResult = detector()
```

Comment (RaeesBhatti) [reply src/utils/detect-server.js:27]: Is it compatible with Node 8?

File: src/commands/dev/index.js (hunk 1 above)
Comment (RaeesBhatti) [reply src/commands/dev/index.js:531]: I don't have prettier setup for `oncommit` hook, it messes up the staged hunks

**Comment by ehmicky** [comment]
Looking good except some small details!
Also, it would be good to fix CI tests because they seem to randomly fail currently (not related to this PR).

**Pull request merged**